### PR TITLE
Remove some unused code

### LINKED ISSUE #57096: The `-Znll-subminimal-causes` option is unused
https://github.com/rust-lang/rust/search?q=nll_subminimal_causes shows only:

https://github.com/rust-lang/rust/blob/0d4f91905bf3071354e30c85f931d11bf2c0009a/src/librustc/session/config.rs#L1322-L1323

The only usage of the option is in the test `dropck_vec_cycle_checked.rs`:

https://github.com/rust-lang/rust/blob/b7c6e8f1805cd8a4b0a1c1f22f17a89e9e2cea23/src/test/ui/span/dropck_vec_cycle_checked.rs#L11

### PR DESCRIPTION
**Pull request by matthewjasper**
Closes #57096

### CODE REVIEW

File: src/librustc/mir/interpret/mod.rs [hunk 1]
```diff
@@ -40,27 +39,6 @@ use ty::codec::TyDecoder;
 use std::sync::atomic::{AtomicU32, Ordering};
 use std::num::NonZeroU32;
 
-#[derive(Clone, Debug, PartialEq, Eq, Hash, RustcEncodable, RustcDecodable)]
```
Comment (nikomatsakis) [inline src/librustc/mir/interpret/mod.rs]: cc @oli-obk, @RalfJung

**Review by nikomatsakis — APPROVED** [review]

**Comment by nikomatsakis** [comment]
I'm going to assume that those deletions from interpret are ok

File: src/librustc/mir/interpret/mod.rs (hunk 1 above)
Comment (RalfJung) [reply src/librustc/mir/interpret/mod.rs]: Yeah this is dead, seems I forgot to remove it.

**Pull request merged**